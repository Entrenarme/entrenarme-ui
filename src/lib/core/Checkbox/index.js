// @flow
import * as React from 'react';
import MCheckbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import styled from 'styled-components';

import colors from '../../helpers/colors';
import P from '../Text/Paragraph';

const theme = createMuiTheme({
  palette: {
    primary: { main: colors.primary.default },
    secondary: { main: colors.secondary.default },
  },
});

type Props = {
  label?: ?string,
};

const SCheckbox = styled(MCheckbox)`
  height: 20px !important;
  width: 20px !important;
  margin-right: 10px !important;
  margin-left: 14px !important;
  svg {
    height: 20px;
  }
`;

const Checkbox = ({ label, ...rest }: Props) => (
  <MuiThemeProvider theme={theme}>
    <FormControlLabel
      control={<SCheckbox {...rest} />}
      label={label ? <P noMargin>{label}</P> : null}
    />
  </MuiThemeProvider>
);

Checkbox.defaultProps = {
  label: null,
};

export default Checkbox;
