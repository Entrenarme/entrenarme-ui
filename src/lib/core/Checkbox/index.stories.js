// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Checkbox from './index';

const actions = {
  onClick: action('onClick'),
};

storiesOf('Checkbox', module)
  .add('default', () => <Checkbox {...actions} />)
  .add('with label', () => <Checkbox {...actions} label="Label" />);
