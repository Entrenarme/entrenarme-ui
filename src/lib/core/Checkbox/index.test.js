import * as React from 'react';
import { render } from 'react-testing-library';

import Checkbox from './index';

test('renders default', () => {
  const { container } = render(<Checkbox />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders with label', () => {
  const { container } = render(<Checkbox label="Label" />);
  expect(container.firstChild).toMatchSnapshot();
});
