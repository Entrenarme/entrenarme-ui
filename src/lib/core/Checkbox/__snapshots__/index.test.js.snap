// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<label
  class="MuiFormControlLabel-root-1"
>
  <span
    class="MuiButtonBase-root-19 MuiIconButton-root-13 MuiSwitchBase-root-9 MuiCheckbox-root-4 MuiCheckbox-colorSecondary-8 sc-bwzfXH jjwSZa"
  >
    <span
      class="MuiIconButton-label-18"
    >
      <svg
        class="MuiSvgIcon-root-22"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        class="MuiSwitchBase-input-12"
        type="checkbox"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-29"
    />
  </span>
  <span
    class="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-3"
  />
</label>
`;

exports[`renders with label 1`] = `
<label
  class="MuiFormControlLabel-root-60"
>
  <span
    class="MuiButtonBase-root-78 MuiIconButton-root-72 MuiSwitchBase-root-68 MuiCheckbox-root-63 MuiCheckbox-colorSecondary-67 sc-bwzfXH jjwSZa"
  >
    <span
      class="MuiIconButton-label-77"
    >
      <svg
        class="MuiSvgIcon-root-81"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        class="MuiSwitchBase-input-71"
        type="checkbox"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-88"
    />
  </span>
  <span
    class="MuiTypography-root-95 MuiTypography-body1-104 MuiFormControlLabel-label-62"
  >
    <p
      class="sc-bdVaJa dnyxcA"
    >
      Label
    </p>
  </span>
</label>
`;
